src/Integration.Vsix/InfoBar/InfoBarManager.cs

```diff
@@ -227,10 +290,23 @@ public PrivateInfoBarWrapper(IVsWindowFrame frame, IVsInfoBarUIElement uiElement
                 this.Advise();
             }
 
+            public PrivateInfoBarWrapper(IVsInfoBarHost host, IVsInfoBarUIElement uiElement)
+            {
+                Debug.Assert(host != null);
+                Debug.Assert(uiElement != null);
+
+                this.Host = host;
+                this.InfoBarUIElement = uiElement;
+
+                this.Advise();
+            }
+
             internal IVsInfoBarUIElement InfoBarUIElement { get; }
 
             internal IVsWindowFrame Frame { get; }
 
+            internal IVsInfoBarHost Host { get; }
```

Comment:
There's a design smell here: the way the class is constructed means only one of the fields will ever have a value, and the caller needs to be aware of this to use the class properly. The refactoring suggested above removes/hides this.

The wrapper class could probably be refactored so that it only accepts a `host`, since in the non-Main window case we still need to get the host via the frame. However, that would involve refactoring the existing code further.